PowerShell/ScubaGear/CheckVersion.ps1

```diff
@@ -0,0 +1,61 @@
+function Invoke-CheckScubaGearVersionPSGallery {
+    # Define the path to the file where we store the last version check time
+    $TempDir = if ($env:TEMP) { $env:TEMP } else { "/tmp" }
+    $VersionCheckFile = [System.IO.Path]::Combine($TempDir, "ScubaVersionCheck.txt")
+
+    # Check if the version check file exists
+    if (Test-Path $VersionCheckFile -ErrorAction 'Stop') {
+        # Read the last check time
+        $LastCheckTime = Get-Content $VersionCheckFile -ErrorAction 'Stop' | Out-String -ErrorAction 'Stop'
+        $LastCheckTime = [datetime]::Parse($LastCheckTime)
+
+        # If the last check was within 24 hours, skip the version check
+        if ((Get-Date -ErrorAction 'Stop') -lt $LastCheckTime.AddHours(24)) {
+            # Exit function without checking for a new version
+            return
+        }
+    }
+
+    # Retrieve the installed version of ScubaGear from the system
+    $InstalledModule = Get-Module -Name ScubaGear -ListAvailable -ErrorAction 'Stop'
+    if ($InstalledModule) {
+        $CurrentVersion = [System.Version]$InstalledModule.Version
+    } else {
+        # If we are here, ScubaGear is not installed from PSGallery.
+        # Or it may have been installed a different way in a nonstandard folder,
+        # or is running in an extracted release folder. Check github instead.
+        return Invoke-CheckScubaGearVersionGithub -ErrorAction 'Stop'
+    }
+
+    # Retrieve the latest version from PowerShell Gallery
+    $ModuleInfo = Find-Module -Name ScubaGear -ErrorAction 'Stop'
+    $LatestVersion = [System.Version]$ModuleInfo.Version
+
+    if ($CurrentVersion -lt $LatestVersion) {
+        Write-Warning "A new version of ScubaGear ($LatestVersion) is available on PowerShell Gallery. This notification can be disabled by setting `$env:SCUBAGEAR_SKIP_VERSION_CHECK = `$true before running ScubaGear."
+
+    }
+
+    # Store the current time in the file to mark the last check time
+    (Get-Date -ErrorAction 'Stop').ToString() | Set-Content $VersionCheckFile -ErrorAction 'Stop'
```

Comment:
With this line here, this file won't be created if you aren't using PSGallery, since the function returns at line 27 otherwise. Do we care about that? I do find it a inconsistent that `VersionCheckFile` would only be used if you use PSGallery, but I don't really care that much. Honestly, with the change to make the check happen on import instead of on invoke, the temp file isn't as important.